### PR TITLE
[MU4] Added draw buffer serialization to json

### PR DESCRIPTION
**Pull request by igorkorsukov**
see also https://github.com/musescore/MuseScore/pull/7614

### CODE REVIEW

File: src/appshell/qml/NotationPage/NotationPage.qml [hunk 1]
```diff
@@ -31,6 +31,8 @@ DockPage {
         pageModel.isNoteInputBarVisible = notationNoteInputBar.visible
         notationNoteInputBar.visible = Qt.binding(function() { return pageModel.isNoteInputBarVisible })
 
+        autobotPanel.visible = Qt.binding(function() { return pageModel.isInspectorPanelVisible })
```
Comment (RomanPudashkin) [inline src/appshell/qml/NotationPage/NotationPage.qml:34]: >isInspectorPanelVisible

Seems this is a typo

File: src/autobot/internal/steps/abdrawserializationstep.cpp [hunk 2]
```diff
@@ -18,15 +18,37 @@
...
+            LOGE() << "failed make path: " << drawDataPath;
+            doFinish(ctx);
+            return;
+        }
+    }
+
+    QByteArray data = AbPaintProvider::instance()->serialize();
+
+    io::path scorePath = ctx.val<io::path>(AbContext::Key::ScoreFile);
+    io::path filePath = drawDataPath + "/" + io::basename(scorePath) + ".json";
```
Comment (RomanPudashkin) [inline src/autobot/internal/steps/abdrawserializationstep.cpp:43]: I think it is better to move this code to the configuration

File: src/autobot/internal/draw/abpaintprovider.cpp [hunk 3]
```diff
@@ -38,14 +39,15 @@ void AbPaintProvider::beginTarget(const std::string& name)
...
 }
 
-bool AbPaintProvider::endTarget(const std::string& name, bool endDraw)
+bool AbPaintProvider::endTarget(bool endDraw)
 {
-    UNUSED(endDraw);
-    LOGI() << name;
-    return BufferedPaintProvider::endTarget(name, endDraw);
+    bool ok = BufferedPaintProvider::endTarget(endDraw);
+    if (ok && buffer().name == "notationview") {
```
Comment (RomanPudashkin) [inline src/autobot/internal/draw/abpaintprovider.cpp:45]: Please explain why you need this check?

File: src/autobot/internal/steps/abdrawserializationstep.cpp (hunk 2 above)
Comment (igorkorsukov) [reply src/autobot/internal/steps/abdrawserializationstep.cpp:43]: I don't think so, this is the path for the current score data, depends on the current context

File: src/autobot/internal/draw/abpaintprovider.cpp (hunk 3 above)
Comment (igorkorsukov) [reply src/autobot/internal/draw/abpaintprovider.cpp:45]: In normal, one paint provider instance for one painter.
In this test, one paint provider instance for all painters (that is, in fact, it is a singleton). This name contains what is currently being painted (there may also be palettes, export, etc.)
In this case, we need only notation view painting

File: src/appshell/qml/NotationPage/NotationPage.qml (hunk 1 above)
Comment (igorkorsukov) [reply src/appshell/qml/NotationPage/NotationPage.qml]: removed

**Review by RomanPudashkin — APPROVED** [review]

**Pull request merged**